### PR TITLE
Use RunAsync and plumb its errCh through

### PR DESCRIPTION
**Pull request by akalin-keybase**
Do this instead of AddCloseListener.

### CODE REVIEW

File: rpc/user_server.go [hunk 1]
```diff
@@ -77,7 +78,9 @@ func (s *perUIDServer) serve() {
 }
 
 func (s *perUIDServer) addConn(a *connectionArgs) error {
-	a.c.xprt.AddCloseListener(s.closeListenCh)
+	go func() {
+		s.closeListenCh <- <-a.c.errCh
```
Comment (akalin-keybase) [inline rpc/user_server.go]: This is a bit weird, but I can't think of a easy way to avoid this.

**Comment by akalin-keybase** [comment]
This is what using https://github.com/keybase/go-framed-msgpack-rpc/pull/72 would look like.

**Comment by maxtaco** [comment]
I'll pass the buck to @patrickxb

**Comment by patrickxb** [comment]
Can you rebase (for diff + CI)?  Changes related to this just went in to master...

**Comment by akalin-keybase** [comment]
Rebased!

**Comment by akalin-keybase** [comment]
Okay, so CI is failing because I haven't checked in the rpc-side PR yet. I'll check that in first once this is LGTMed and wait until CI passes until merging this.

**Comment by patrickxb** [comment]
Well, it's just a little hard for me to LGTM this if I can't test it...

**Comment by akalin-keybase** [comment]
Okay, let me merge the other one. (I was holding off until I got PRs ready for client and KBFS-side changes, but those won't be too bad, I think.)

**Comment by akalin-keybase** [comment]
Okay, merged the other one. FYI, you could still have tested it when it wasn't merged by checking out the rpc repo and switching the branch there.

File: rpc/user_server.go [hunk 2]
```diff
@@ -86,23 +87,15 @@ func (s *perUIDServer) serve() {
 }
 
 func (s *perUIDServer) addConn(a *connectionArgs) error {
-
-	a.c.xprt.AddCloseListener(s.closeListenCh)
-
+	go func() {
+		s.closeListenCh <- <-a.c.errCh
+	}()
```
Comment (patrickxb) [inline rpc/user_server.go]: Is this a possible goroutine leak?  Are we guaranteed to receive something from `a.c.errCh`?

Comment (akalin-keybase) [inline rpc/user_server.go]: Yeah -- it'll be sent on when the RPC connection is closed for whatever reason.

I think I may be able to get rid of the need for this by making more changes to the RPC library (which I'm planning to do), but I don't want to change too much at once...

Comment (maxtaco) [inline rpc/user_server.go]: How does this compile?  Shouldn't it be `s.closeListenCh`?

Comment (akalin-keybase) [inline rpc/user_server.go]: It's like:
```
t := <-a.c.errCh // t has type error
s.closeListenCh <- t
```

Comment (maxtaco) [inline rpc/user_server.go]: I think I'm referring to a GH rendering error!?!?!

Comment (maxtaco) [inline rpc/user_server.go]: screen shot 2016-04-08 at 6 14 59 pm

Comment (akalin-keybase) [inline rpc/user_server.go]: Yeah, GitHub is weird with that sometimes. Note the weird indentation, too! But I've never seen it hide characters before...

Comment (maxtaco) [inline rpc/user_server.go]: Unified view shows the right thing. Sorry 'bout that.

**Comment by akalin-keybase** [comment]
BTW, I'm reverting the RPC-side change, since gregor master can't build. (I was expecting a quicker LGTM...) @patrickxb if you want to test it more you'll have to do the procedure I described above (should be easy).

**Comment by maxtaco** [comment]
LGTM if ci is green

**Pull request merged**